sklearnex/spmd/basic_statistics/tests/test_incremental_basic_statistics_spmd.py

```diff
@@ -0,0 +1,291 @@
+    dataframe, queue, num_blocks, weighted, result_option, dtype
+):
+    # Import spmd and batch algo
+    from sklearnex.basic_statistics import IncrementalBasicStatistics
+    from sklearnex.spmd.basic_statistics import (
+        IncrementalBasicStatistics as IncrementalBasicStatistics_SPMD,
+    )
+
+    # Create gold data and process into dpt
+    data = np.array(
+        [
+            [0.0, 0.0, 0.0],
+            [0.0, 1.0, 2.0],
+            [0.0, 2.0, 4.0],
+            [0.0, 3.0, 8.0],
+            [0.0, 4.0, 16.0],
+            [0.0, 5.0, 32.0],
+            [0.0, 6.0, 64.0],
+            [0.0, 7.0, 128.0],
+        ],
+        dtype=dtype,
+    )
+    dpt_data = _convert_to_dataframe(data, sycl_queue=queue, target_df=dataframe)
+    local_data = _get_local_tensor(data)
+    split_local_data = np.array_split(local_data, num_blocks)
+
+    if weighted:
+        weights = np.array([0.0, 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0], dtype=dtype)
+        dpt_weights = _convert_to_dataframe(
+            weights, sycl_queue=queue, target_df=dataframe
+        )
+        local_weights = _get_local_tensor(weights)
+        split_local_weights = np.array_split(local_weights, num_blocks)
+
+    option, _, _ = result_option
+    incbs_spmd = IncrementalBasicStatistics_SPMD(result_options=option)
+    incbs = IncrementalBasicStatistics(result_options=option)
+
+    for i in range(num_blocks):
+        local_dpt_data = _convert_to_dataframe(
+            split_local_data[i], sycl_queue=queue, target_df=dataframe
+        )
+        if weighted:
+            local_dpt_weights = _convert_to_dataframe(
+                split_local_weights[i], sycl_queue=queue, target_df=dataframe
+            )
+        incbs_spmd.partial_fit(
+            local_dpt_data, sample_weight=local_dpt_weights if weighted else None
+        )
+
+    incbs.fit(dpt_data, sample_weight=dpt_weights if weighted else None)
+
+    assert_allclose(getattr(incbs_spmd, option), getattr(incbs, option))
+
+
+@pytest.mark.skipif(
+    not _mpi_libs_and_gpu_available,
+    reason="GPU device and MPI libs required for test",
+)
+@pytest.mark.parametrize(
+    "dataframe,queue",
+    get_dataframes_and_queues(dataframe_filter_="dpnp,dpctl", device_filter_="gpu"),
+)
+@pytest.mark.parametrize("num_blocks", [1, 2])
+@pytest.mark.parametrize("weighted", [True, False])
+@pytest.mark.parametrize("n_samples", [100, 10000])
+@pytest.mark.parametrize("n_features", [10, 100])
+@pytest.mark.parametrize("dtype", [np.float32, np.float64])
+@pytest.mark.mpi
+def test_incremental_basic_statistics_partial_fit_spmd_synthetic(
+    dataframe, queue, num_blocks, weighted, n_samples, n_features, dtype
+):
+    # Import spmd and batch algo
+    from sklearnex.basic_statistics import IncrementalBasicStatistics
+    from sklearnex.spmd.basic_statistics import (
+        IncrementalBasicStatistics as IncrementalBasicStatistics_SPMD,
+    )
+
+    tol = 2e-3 if dtype == np.float32 else 1e-7
```

Comment:
Woah, is difference in tolerance 4 orders of magnitude? If so then please make a ticket for investigation, this is really really bad.

Reply:
ticket 8350